automation/taskcluster/release/convert_group_and_artifact_ids.py

```diff
@@ -0,0 +1,264 @@
+        new_file_name = convert_file_name(
+            file_name, old_group_id, old_artifact_id, new['new_group_id'], new['new_artifact_id']
+        )
+        log.debug('File "{}" renamed into "{}"'.format(file_name, new_file_name))
+    except KeyError:
+        new_file_name = file_name
+        log.debug('Keeping old file name "{}"'.format(new_file_name))
+
+    return new_file_name
+
+
+def convert_file_name(file_name, old_group_id, old_artifact_id, new_group_id, new_artifact_id):
+    return file_name.replace(
+        old_group_id.replace('.', '/'), new_group_id.replace('.', '/')
+    ).replace(
+        old_artifact_id, new_artifact_id
+    )
+
+
+def extract_group_id(file_name):
+    return '.'.join(file_name.split('/')[:-3])
+
+
+def extract_artifact_id(file_name):
+    return file_name.split('/')[-3]
+
+
+def process_pom_and_checksums(zip_in, zip_out, zip_info, new_file_name, conversion_dict):
+    raw_xml = zip_in.read(zip_info.filename)
+    pom = translate_pom_xml(raw_xml, conversion_dict)
+    zip_out.writestr(new_file_name, pom, compress_type=zipfile.ZIP_DEFLATED)
+
+    for hash_alg in SUPPORTED_HASHES:
+        sum = get_checksum(pom, hash_alg)
+        zip_out.writestr('{}.{}'.format(new_file_name, hash_alg), sum, compress_type=zipfile.ZIP_DEFLATED)
+
+
+def copy_file_into_zip(zip_in, zip_out, zip_info, new_file_name):
+    with tempfile.NamedTemporaryFile() as temp:
+        temp.write(zip_in.read(zip_info.filename))
+        temp.seek(0)
+        zip_out.write(
+            temp.name, arcname=new_file_name,
+            # TODO: Figure out why zip_info.compress_type is invalid
+            compress_type=zipfile.ZIP_DEFLATED
+        )
+
+
+def get_checksum(data, hash_alg):
+    h = hashlib.new(hash_alg)
+    h.update(data)
+    return h.hexdigest()
+
+
+def translate_pom_xml(raw_xml, conversion_dict):
+    root = ElementTree.fromstring(raw_xml)
+    convert_xml_element(root, conversion_dict)
+
+    dependencies = root.find('m:dependencies', XML_NAMESPACES)
+    if dependencies is not None:    # Some packages may not have any dependencies
+        for dependency in dependencies.findall('m:dependency', XML_NAMESPACES):
+            convert_xml_element(dependency, conversion_dict)
+
+    new_xml = ElementTree.tostring(root, encoding='UTF-8')
+    log.debug('pom.xml modified. From: {}\n\nTo: {}'.format(raw_xml, new_xml))
+
+    return new_xml
+
+
+def convert_xml_element(element, conversion_dict):
+    group_id_element = element.find('m:groupId', XML_NAMESPACES)
+    old_group_id = group_id_element.text
+    artifact_id_element = element.find('m:artifactId', XML_NAMESPACES)
+    old_artifact_id = artifact_id_element.text
+
+    try:
+        new = conversion_dict[get_conversion_key(old_group_id, old_artifact_id)]
+        group_id_element.text = new['new_group_id']
+        artifact_id_element.text = new['new_artifact_id']
+    except KeyError:
+        pass
```

Comment:
Should we indeed `pass` here or should we raise something? All the modules will have these two fields set in their `upstreamZip` so a `KeyError`  might mean something else or come from elsewhere.

Reply:
Neah, worry not, it's clear now. This is temporary anyway so I don't think it's worth investing time in more optimizations. Hopefully we can retire this as soon as possible to rely solely on information from `settings.gradle` instead.